src/Middlewares/PermissionMiddleware.php

```diff
@@ -13,14 +13,14 @@ public function handle($request, Closure $next, $permission)
             abort(403);
         }
 
-        $permission = is_array($permission)
+        $permissions = is_array($permission)
             ? $permission
             : explode('|', $permission);
 
-        if (! Auth::user()->hasAnyPermission(...$permission)) {
-            abort(403);
+        if (Auth::user()->can(...$permissions)) {
```

Comment:
Does this line work?

From the docs:

```
if ($user->can('create', Post::class)) {
    // Executes the "create" method on the relevant policy...
}
```

The second parameter should be a model, no mention about next parameters.

Reply:
Based on the code the second parameter is optional.

https://github.com/laravel/framework/blob/5.5/src/Illuminate/Foundation/Auth/Access/Authorizable.php

Reply:
The tests are passing. Not sure if there are additional tests needed to prove the middleware is firing correctly. 

I have the code working on a production site, and it appears to be properly functional.

Reply:
I'm thinking the middleware will now only work by chance if there is only one permission passed. It'll not work correctly when 2 or more permissions are passed and the 2nd, 3th, ... parameters is an ability the user can do. 

```php
if (Auth::user()->can(...$permissions)) {
```

should be changed to 

```php
if (Auth::user()->can($permissions)) {
```

Best to add some tests to prove that my thinking is correct/wrong 😄

Reply:
In playing with the tests, it appears you are correct :)

I've added some more tests: passing an array (even though most middleware calls will be string and not array), and re-ordering options so that the first option isn't the only one under test.